src/callback.rs

```diff
@@ -35,6 +35,30 @@ impl PyCallbackOutput for () {
     const ERR_VALUE: Self = ();
 }
 
+#[doc(hidden)]
+pub trait IntoResult<R> {
+    fn into_result(self) -> R;
+}
+
+impl<T> IntoResult<PyResult<T>> for T
```

Comment:
Do you think the trait should potentially just be `IntoPyResult<T>`, and the method `into_py_result()`?

That'll have the effect of making error messages clearly related to PyO3.

Reply:
Just to clarify - the only allowed return types for pyo3 functions and methods are:

```
Result<T, E>
where
    T: IntoPy<PyObject>,
    E: Into<PyErr>,
```
and
```
T
where
    T: IntoPy<PyObject>,
```
right?

I don't see it anywhere written in the guide, and CI doesn't seem to complain, but I'd like to make sure.

Reply:
Yes (except for `#[new]` and magic methods like `__hash__` which return specific types).